spec/unit/functions/lookup_spec.rb

```diff
@@ -416,6 +416,19 @@ def compile_and_get_notifications(code)
     end
   end
 
+  context 'when accessing empty files' do
+    # An empty YAML file is OK and should be treated as a file that contains no keys
+    it "will fail normally with a 'did not find a value' error when a yaml file is empty" do
+      Puppet[:code] = "include empty_yaml\nlookup('empty_yaml::a')"
+      expect { compiler.compile }.to raise_error(Puppet::ParseError, /did not find a value for the name 'empty_yaml::a'/)
+    end
+
+    # An empty JSON file is not OK. Should yield a parse error
+    it "will fail with a LookupError indicating a parser failure when a json file is empty" do
```

Comment:
Because empty does not represent anything in JSON, not even null. Good answer with links here if you want more details: http://stackoverflow.com/questions/18419428/what-is-the-minimum-valid-json